1_IGT_PP/Code/R/0_Preprocess/IGT_Preprocess.R

```diff
@@ -392,11 +394,14 @@ for (i in 1:numSubjs) {
       card[i,,s] <- tmp_dat$stim
       RLmatrix[i,,s]  <- tmp_dat$rewlos
       SRLmatrix[i,,s] <- tmp_dat$Srewlos
+      IDs[i,,s] <- tmp_dat$Subject
     }
   }
 }
 
-
+# Check that how the for loop above loops through the subject IDs is in the same order as subjList
+(condensedIDs = apply(IDs, c(1,3), mean)) # Visually check they are the same (-1 means missing data on that session)
+ifelse(mean(condensedIDs[,1] == subjList) == 1, "Equal", "Not Equal") # Check if IDs in for loop equal IDs in subjList
```

Comment:
Instead of this, you can just explicitly check that they are all identical using the following syntax (makes the intention a little more clear IMO): 

```suggestion
ifelse(all(stan_data$igt_IDs == stan_data$sr_IDs), "Equal", "Not Equal") # Check if IDs in for loop equal IDs in subjList
```